src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -195,31 +169,18 @@ export function createModelForService(
             const apiVersionIndex = op.Parameters.findIndex(
                 (value: InputParameter) => value.IsApiVersion
             );
-            if (apiVersionIndex !== -1) {
-                const apiVersionInOperation = op.Parameters[apiVersionIndex];
+            if (apiVersionIndex === -1) {
+                continue;
+            }
+            const apiVersionInOperation = op.Parameters[apiVersionIndex];
+            if (defaultApiVersionConstant !== undefined) {
                 if (!apiVersionInOperation.DefaultValue?.Value) {
                     apiVersionInOperation.DefaultValue =
-                        apiVersionParam.DefaultValue;
-                }
-                /**
-                 * replace to the global apiVersion parameter if the apiVersion defined in the operation is the same as the global service apiVersion parameter.
-                 * Three checkpoints:
-                 * the parameter is query parameter,
-                 * it is client parameter
-                 * it does not has default value, or the default value is included in the global service apiVersion.
-                 */
-                if (
-                    apiVersions.has(
-                        apiVersionInOperation.DefaultValue?.Value
-                    ) &&
-                    apiVersionInOperation.Kind ===
-                        InputOperationParameterKind.Client &&
-                    apiVersionInOperation.Location === apiVersionParam.Location
-                ) {
-                    op.Parameters[apiVersionIndex] = apiVersionParam;
+                        defaultApiVersionConstant;
                 }
             } else {
-                op.Parameters.push(apiVersionParam);
+                apiVersionInOperation.Kind = InputOperationParameterKind.Method;
+                apiVersionInOperation.IsApiVersion = false;
```

Comment:
Why set the `IsApiVersion` false? The parameter is api-version.

Reply:
It is unversioned scenario, so it is not.

Reply:
The parameter is api-version parameter, but it is not get the value from clientOptions. Maybe in autorest.csharp codegen, it will use `IsApiVersion` flag to do some special handling.